src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
+const ANIMAL_HELTH = 100;
+
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = ANIMAL_HELTH;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide(isHidden = true) {
+    this.hidden = isHidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    const BITE_DMG = 50;
+
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= BITE_DMG;
+    }
+
+    if (!target.health) {
+      Animal.alive = Animal.alive.filter(animal => animal.health > 0);
```

Comment:
it's better to create a static method for the class that will delete the animal:

```
if (target.health <= 0) {
  Animal.kill(victim);
}
```